### PR TITLE
Remove the requirement of down.sh for arguments

### PR DESCRIPTION
**Pull request by letzya**
1. Register the deployments into ./.bootstrap/bootstrapped-deployments during up.sh
2. Read from ./.bootstrap/bootstrapped-deployments during down.sh
3. Added printouts to test if an error has occurred (?$)
4. Added printout of the docker-compose command as I use it very often and since here it's complicated due to many directories I thought printing it to the user would be useful.
(Example: `docker-compose -f deployments/tyk/docker-compose.yml -f deployments/sso/docker-compose.yml -f deployments/tls/docker-compose.yml -p tyk-pro-docker-demo-extended --project-directory /Users/yaara/work/src/github.com/TykTechnologies/tyk-demo ps`)
5. Have updated the README accordingly
6. Added `--remove-orphanes` to remove containers for services not defined in the Compose file.

### CODE REVIEW

**Comment by letzya** [comment]
@davegarvey I can't assign you as a reviewer.

**Comment by davegarvey** [comment]
@letzya ok, I've updated the permissions, the engineering team now have write access. I'd still like to follow a PR process for now though.

File: down.sh [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+done < $bootstrapped_deployments_file
+fi
 command_docker_compose="$command_docker_compose -p tyk-pro-docker-demo-extended --project-directory $(pwd) down -v"
-eval $command_docker_compose
\ No newline at end of file
+echo "Running docker-compose down: " $command_docker_compose
+eval $command_docker_compose
+if [ "$?" == 0 ]
+then
+  echo "All containers were stoped and removed"
```
Comment (davegarvey) [inline down.sh:21]:
```suggestion
  echo "All containers were stopped and removed"
```

File: up.sh [hunk 2]
```diff
@@ -56,19 +59,41 @@ do
...
   fi
 done
-command_docker_compose="$command_docker_compose -p tyk-pro-docker-demo-extended --project-directory $(pwd) up -d"
+
+echo "Starting containers: $command_docker_compose"
+command_docker_compose="$command_docker_compose -p tyk-pro-docker-demo-extended --project-directory $(pwd) up --remove-orphans -d"
 eval $command_docker_compose
+if [ "$?" != 0 ]
+then
+  echo "Failure occured when started docker-compose"
```
Comment (davegarvey) [inline up.sh:68]:
```suggestion
  echo "Error occurred when using docker-compose to bring containers up"
```

File: up.sh [hunk 3]
```diff
@@ -56,19 +59,41 @@ do
...
+  exit
+fi
 
 # alway run the tyk bootstrap first
-deployments/tyk/bootstrap.sh
+echo "bootstrapping 'tyk' deployment"
+deployments/tyk/bootstrap.sh 2>> bootstrap.log
+if [ "$?" != 0 ]
+then
+  echo "Failure during bootstrap of 'tyk'. For details check bootstrap.log"
```
Comment (davegarvey) [inline up.sh:77]:
```suggestion
  echo "Error occurred during bootstrap of 'tyk' deployment. Check bootstrap.log for details."
```

File: up.sh [hunk 4]
```diff
@@ -56,19 +59,41 @@ do
...
 for var in "$@"
 do
   # the `tyk` deployment is already included, so don't duplicate it
   if [ "$var" != "tyk" ]
   then
+    echo "bootstrapping: deployments/$var/bootstrap.sh"
     eval "deployments/$var/bootstrap.sh"
+    if [ "$?" != 0 ]
+    then
+      echo "Failure during bootstrap of $var (check deployments/$var/bootstrap.sh). For details check bootstrap.log"
```
Comment (davegarvey) [inline up.sh:91]:
```suggestion
      echo "Error occurred during bootstrap of $var, when running deployments/$var/bootstrap.sh. Check bootstrap.log for details."
```

**Pull request merged**